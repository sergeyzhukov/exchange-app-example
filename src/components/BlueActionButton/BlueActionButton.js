import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

export default ({ title, onPress, style, disabled = false }) => (
  <TouchableOpacity
    style={[styles.container, style, disabled && styles.disabled]}
    onPress={onPress}
    disabled={disabled}
  >
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#4072B8',
    height: 44,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 2,
  },
  title: {
    color: 'white',
    fontSize: 16,
  },
  disabled: {
    opacity: 0.3,
  },
})
