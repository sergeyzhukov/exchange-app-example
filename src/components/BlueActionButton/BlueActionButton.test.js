import React from 'react'
import renderer from 'react-test-renderer'
import BlueActionButton from './BlueActionButton'

test('renders correctly', () => {
  const component = <BlueActionButton />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly width disabled', () => {
  const component = <BlueActionButton disabled />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
