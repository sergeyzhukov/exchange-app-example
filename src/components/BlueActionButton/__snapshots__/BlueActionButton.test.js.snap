// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4072B8",
      "borderRadius": 2,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  />
</View>
`;

exports[`renders correctly width disabled 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4072B8",
      "borderRadius": 2,
      "height": 44,
      "justifyContent": "center",
      "opacity": 0.3,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  />
</View>
`;
